app/services/streamlabels/index.ts

```diff
@@ -0,0 +1,212 @@
+    if (settings.item_separator) {
+      settings.item_separator = settings.item_separator.replace(/\\n/gi, '\n');
+    }
+
+    this.settings[statname] = settings;
+
+    const headers = new Headers();
+    headers.append('Content-Type', 'application/json');
+
+    const url = `https://${this.hostsService.streamlabs}/api/v5/slobs/stream-labels` +
+    `/settings?token=${this.userService.widgetToken}`;
+    const request = new Request(url, {
+      method: 'POST',
+      headers,
+      body: JSON.stringify(this.settings)
+    });
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(() => true);
+  }
+
+
+  /**
+   * Attempt to load initial data via HTTP instead of waiting
+   * for a socket event
+   */
+  fetchInitialData() {
+    if (!this.userService.isLoggedIn()) return;
+
+    const url = `https://${this.hostsService.streamlabs}/api/v5/slobs/stream-labels` +
+      `/files?token=${this.userService.widgetToken}`;
+    const request = new Request(url);
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json())
+      .then(json => this.setStreamlabelsData(json.data));
+  }
+
+
+  fetchSettings(): Promise<Dictionary<IStreamlabelSettings>> {
+    if (!this.userService.isLoggedIn()) return Promise.reject({});
+
+    const url = `https://${this.hostsService.streamlabs}/api/v5/slobs/stream-labels` +
+      `/settings?token=${this.userService.widgetToken}`;
+    const request = new Request(url);
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json());
+  }
+
+
+  fetchSocketToken(): Promise<string> {
+    if (!this.userService.isLoggedIn()) return Promise.reject('User must be logged in');
+
+    const url = `https://${this.hostsService.streamlabs}/api/v5/slobs/socket-token/${this.userService.widgetToken}`;
+    const request = new Request(url);
+
+    return fetch(request)
+      .then(handleErrors)
+      .then(response => response.json())
+      .then(json => json.socket_token);
+  }
+
+
+  setStreamlabelsData(data: IStreamlabelsData) {
+    this.data = { ...data };
+
+    this.subscriptions.forEach(subscription => this.updateSubscription(subscription));
+  }
+
+
+  private ensureDirectory() {
+    if (fs.existsSync(this.streamlabelsDirectory)) {
+      rimraf.sync(this.streamlabelsDirectory);
+    }
+
+    fs.mkdirSync(this.streamlabelsDirectory);
+  }
+
+
+  private get streamlabelsDirectory() {
+    return path.join(electron.remote.app.getPath('userData'), 'Streamlabels');
+  }
+
+
+  getStreamlabelsPath(filename: string) {
+    return path.join(this.streamlabelsDirectory, `${filename}.txt`);
+  }
+
+
+  /**
+   * Updates the data in a single streamlabels file
+   * @param filename the filename
+   * @param statname the name of the stat
+   */
+  updateSubscription(subscription: IStreamlabelsSubscription) {
+    if (!this.data) return;
+
+    console.log('Writing file', subscription.filename, subscription.statname);
```

Comment:
should be `this.log()`

Reply:
Actually meant to remove this one entirely.